spec/features/consumer/account/settings_spec.rb

```diff
@@ -27,5 +31,18 @@
       click_link I18n.t('spree.users.show.tabs.settings')
       expect(page).to have_content I18n.t('spree.users.show.unconfirmed_email', unconfirmed_email: 'new@email.com')
     end
+
+    it "allows the user to change their password" do
+      allow(Spree::Auth::Config).to receive(:[]).with(:signout_after_password_change).and_return(false)
```

Comment:
@mkllnk I've added this to make the spec work, as the test environment doesn't pick up the value set in the spree initializer. Do you think this is ok, or should I try to set that config variable globally in the test environment instead? I'm wondering if it might interfere with other specs or changes to factories if it's set globally...

Reply:
That's the way IMO. Changes to global state in tests => :boom:

Reply:
I agree that changes to the global state are not good in tests in general. And I find this solution okay, but I see a problem in the fact that the logic is different for production and testing here. We have to add this line to the test case to make it behave the same way. That's basically code duplication in a very hidden place. Imagine we had to define the checkout state machine in the same spec that verifies that it's correct. It's not really testing the code used in production then.